File: web/src/api/tasks.ts
import request from './client';

export interface AutomationTask {
  id: number;
  name: string;
  event: string;
  path_pattern?: string;
  filename_regex?: string;
  processor_type: string;
  processor_config: Record<string, any>;
  enabled: boolean;
}

export type AutomationTaskCreate = Omit<AutomationTask, 'id'>;
export type AutomationTaskUpdate = Partial<AutomationTaskCreate>;

export const tasksApi = {
  list: () => request<AutomationTask[]>('/tasks/'),
  create: (payload: AutomationTaskCreate) => request<AutomationTask>('/tasks', { method: 'POST', json: payload }),
  update: (id: number, payload: AutomationTaskUpdate) => request<AutomationTask>(`/tasks/${id}`, { method: 'PUT', json: payload }),
  remove: (id: number) => request<void>(`/tasks/${id}`, { method: 'DELETE' }),
};

File: schemas/tasks.py
from pydantic import BaseModel
from typing import Optional, Dict, Any


class AutomationTaskBase(BaseModel):
    name: str
    event: str
    path_pattern: Optional[str] = None
    filename_regex: Optional[str] = None
    processor_type: str
    processor_config: Dict[str, Any] = {}
    enabled: bool = True


class AutomationTaskCreate(AutomationTaskBase):
    pass


class AutomationTaskUpdate(AutomationTaskBase):
    name: Optional[str] = None
    event: Optional[str] = None
    processor_type: Optional[str] = None
    processor_config: Optional[Dict[str, Any]] = None
    enabled: Optional[bool] = None


class AutomationTaskRead(AutomationTaskBase):
    id: int

    class Config:
        from_attributes = True
